src/agent/src/storage/block_handler.rs

```diff
@@ -82,10 +85,33 @@ impl StorageHandler for VirtioBlkPciHandler {
             if mode & libc::S_IFBLK == 0 {
                 return Err(anyhow!("Invalid device {}", &storage.source));
             }
+            let dev_id = metadata.rdev();
+            dev_num = format!("{}:{}", major(dev_id), minor(dev_id));
         } else {
             let pcipath = pci::Path::from_str(&storage.source)?;
             let dev_path = get_virtio_blk_pci_device_name(ctx.sandbox, &pcipath).await?;
             storage.source = dev_path;
+            let metadata = fs::metadata(&storage.source)
+                .context(format!("get metadata on file {:?}", &storage.source))?;
+            let dev_id = metadata.rdev();
+            dev_num = format!("{}:{}", major(dev_id), minor(dev_id));
+        }
+
+        let confidential = storage
+            .driver_options
+            .contains(&"confidential=true".to_string());
+        let ephemeral = storage
+            .driver_options
+            .contains(&"ephemeral=true".to_string());
+
+        if confidential && ephemeral {
```

Comment:
I don't quite understand, from the Kata perspective, why we're limited to ephemeral volumes here. The CDH should be connected to Trustee already - is there a reason why it can't request stable encryption keys?

Reply:
I haven't yet locked in on a design if/when attestation is driven by a sidecar so I don't want to commit to an implementation here, although this is extensible.